enarx-keep/shims/shim-sev/src/shim_stack.rs

```diff
@@ -5,26 +5,29 @@
 use crate::frame_allocator::FRAME_ALLOCATOR;
 use crate::paging::SHIM_PAGETABLE;
 use core::ops::DerefMut;
-use nbytes::bytes;
 use x86_64::structures::paging::{Page, PageTableFlags, Size4KiB};
 use x86_64::VirtAddr;
 
-/// The virtual address of the main kernel stack
-pub const SHIM_STACK_START: u64 = 0xFFFF_FF48_4800_0000;
-
-/// The size of the main kernel stack
-#[allow(clippy::integer_arithmetic)]
-pub const SHIM_STACK_SIZE: u64 = bytes![4; MiB];
+/// A guarded stack
+pub struct GuardedStack {
+    /// the stack pointer
+    pub pointer: VirtAddr,
+    /// the usable stack memory slice
+    pub slice: &'static mut [u8],
+}
 
-/// Allocate the stack for the shim with guard pages
-#[allow(clippy::integer_arithmetic)]
-pub fn init_shim_stack() -> VirtAddr {
+/// Allocate a stack with guard pages
+pub fn init_stack_with_guard(
+    start: VirtAddr,
+    stack_size: u64,
+    extra_flags: PageTableFlags,
+) -> GuardedStack {
     // guard page
     FRAME_ALLOCATOR
         .write()
         .allocate_and_map_memory(
             SHIM_PAGETABLE.write().deref_mut(),
-            VirtAddr::new(SHIM_STACK_START - Page::<Size4KiB>::SIZE),
+            start - Page::<Size4KiB>::SIZE,
```

Comment:
@haraldh Should this be `start + Page::<Size4KiB>::SIZE`? It looks like this guard page overlaps with the first page of the stack. The stack pointer below will be set to this guard page, rather than the usable stack area.

Reply:
yeah.. it's one page before and one page after the usable stack